source/common/router/router.cc

```diff
@@ -950,7 +950,7 @@ void Filter::onResponseTimeout() {
     // We want to record the upstream timeouts and increase the stats counters in all the cases.
     // For example, we also want to record the stats in the case of BiDi streaming APIs where we
     // might have already seen the headers.
-    cluster_->trafficStats().upstream_rq_timeout_.inc();
+    cluster_->trafficStats()->upstream_rq_timeout_.inc();
```

Comment:
just want to point out that this whole lazy optimization change has a lot of sense from the perspective of the mobile Envoy (that runs on iOS and Android devices) too.

For the mobile applications a lot of impact could be realized if we added lazy initiation to virtual cluster stats. They are managed in multiple places but one of them is just a line below the line I commented on. Are there any plans to make the initialization of virtual cluster stats lazy too or is this outside of the scope of work/changes that you are work on?